react/src/lib/components/WebvizDialog/WebvizDialog.tsx

```diff
@@ -0,0 +1,732 @@
+            }
+
+            const currentMousePosition = {
+                x: Math.max(clientX, 0),
+                y: Math.max(clientY, 0),
+            };
+            const delta = pointDifference(
+                currentMousePosition,
+                prevMousePosition
+            );
+
+            if (!isMoveStarted) {
+                if (vectorLength(delta) > MANHATTAN_LENGTH) {
+                    isMoveStarted = true;
+                }
+            } else {
+                setDialogLeft((prev) => delta.x + prev);
+                setDialogTop((prev) => delta.y + prev);
+                prevMousePosition = currentMousePosition;
+            }
+        };
+
+        const handleBlur = () => {
+            isMoveStarted = false;
+            isMouseDown = false;
+        };
+
+        const handleEscapeKeyUp = (e: KeyboardEvent) => {
+            if (e.key !== "Escape") {
+                return;
+            }
+            dialogRef.current?.classList.contains("WebvizDialog--active") &&
+                handleClose(DialogCloseReason.ESCAPE_BUTTON_PRESSED);
+            e.preventDefault();
+        };
+
+        if (dialogTitleRef.current && !props.disableDraggable) {
+            dialogTitleRef.current.addEventListener(
+                "mousedown",
+                handleMouseDown
+            );
+            dialogTitleRef.current.addEventListener(
+                "touchstart",
+                handleTouchStart
+            );
+        }
+
+        document.addEventListener("mouseup", handleMouseUpAndTouchEnd);
+        document.addEventListener("mousemove", handleMouseMove);
+        document.addEventListener("touchend", handleMouseUpAndTouchEnd);
+        document.addEventListener("touchmove", handleTouchMove);
+        window.addEventListener("blur", handleBlur);
+        if (!props.disableEscapeKeyDown) {
+            window.addEventListener("keyup", handleEscapeKeyUp);
+        }
+
+        return () => {
+            if (dialogTitleRef.current) {
+                dialogTitleRef.current.removeEventListener(
+                    "mousedown",
+                    handleMouseDown
+                );
+                dialogTitleRef.current.removeEventListener(
+                    "touchstart",
+                    handleTouchStart
+                );
+            }
+
+            window.removeEventListener("blur", handleBlur);
+            document.removeEventListener("mousemove", handleMouseMove);
+            document.removeEventListener("mouseup", handleMouseUpAndTouchEnd);
+            document.removeEventListener("touchmove", handleTouchMove);
+            document.removeEventListener("touchend", handleMouseUpAndTouchEnd);
+
+            if (!props.disableEscapeKeyDown) {
+                window.removeEventListener("keyup", handleEscapeKeyUp);
+            }
+        };
+    }, [
+        dialogRef.current,
+        dialogTitleRef.current,
+        props.disableEscapeKeyDown,
+        props.disableDraggable,
+    ]);
```

Comment:
No refs required.

Reply:
`dialogTitleRef.current` kept as it is a `forwardRef`